pegasus/sites.v3/code.org/public/v3/edit-csp-table/splat.haml

```diff
@@ -151,6 +151,21 @@ angular: recordsApp
       column.id = column.name;
       column.editing = false;
     }
+    $scope.coerceColumn = function (column, type) {
+      $http({
+        method:'POST',
+        url: '/v3/coerce-shared-tables/#{channel_id}/#{table_name}',
+        params: {column_name: column.id, type: type}
+      }).then(function (response) {
+        var conversionError = '';
+        if (response.data !== 'true') {
+          // reload the page with conversionError set, allowing us to show an error string
+          conversionError = '?conversionError=Not all values in column ' + column.id +
+            ' could be converted to type ' + type + '.';
+        }
+        location.assign(location.pathname + conversionError);
+      });
```

Comment:
you shouldn't have to reload the page to show an error in the UI. when you set anything on `$scope` it should be available within splat.haml. So I am proposing:
```
+      }).then(function (response) {
 +        var $scope.conversionError = '';
 +        if (response.data !== 'true') {
 +          $scope.conversionError = 'Not all values in column ' + column.id +
 +            ' could be converted to type ' + type + '.';
 +        }
 +      });
```
no changes should be needed to splat.haml.